### PR TITLE
AzurePipelinesCredential | adding mlflow uri func

### PR DESCRIPTION
**Pull request by kshitij-microsoft**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/workspace.py [hunk 1]
```diff
@@ -222,7 +228,22 @@ def mlflow_tracking_uri(self) -> Optional[str]:
...
+        try:
+            from azureml.mlflow import get_mlflow_tracking_uri_v2
+
+            return get_mlflow_tracking_uri_v2(self)
+        except ImportError as e:
+            error_msg = (
+                "azureml.mlflow could not be imported. "
+                "Please ensure that 'azureml-mlflow' has been installed in the current python environment."
+            )
+            raise UserErrorException(error_msg) from e
```
Comment (singankit) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/workspace.py:244]: We should not raise an error message if azureml-mlflow is not installed. It should be a warning message. Return the tracking uri. In warning say azureml-mlflow will not use credentials passed to MLCLient

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/workspace.py [hunk 2]
```diff
@@ -222,7 +228,22 @@ def mlflow_tracking_uri(self) -> Optional[str]:
...
         """
-        return self._mlflow_tracking_uri
+        # if _with_auth:
+        #     module_logger.warning(
+        #     "'_with_auth' is deprecated and will be removed in a future release. ")
+
+        try:
+            from azureml.mlflow import get_mlflow_tracking_uri_v2
+
+            return get_mlflow_tracking_uri_v2(self)
```
Comment (singankit) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/workspace.py:238]: Just FYI we cannot release this until PR in azureml-mlflow goes out.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/workspace.py [hunk 3]
```diff
@@ -329,8 +333,20 @@ def _from_rest_object(cls, rest_obj: RestWorkspace) -> Optional["Workspace"]:
...
-            mlflow_tracking_uri = rest_obj.ml_flow_tracking_uri
+            try:
+                from azureml.mlflow import get_mlflow_tracking_uri_v2
+
+                mlflow_tracking_uri = get_mlflow_tracking_uri_v2(rest_obj, v2_service_context)
+            except ImportError:
+                mlflow_tracking_uri = rest_obj.ml_flow_tracking_uri
+                error_msg = (
+                    "azureml.mlflow could not be imported. azureml-mlflow will not use credentials passed to `MLClient`"
+                    "Please ensure that latest 'azureml-mlflow' has been installed in the current python environment. "
```
Comment (achauhan-scc) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/workspace.py:346]: error message is not clear, customer can have older version of azureml-mlflow where get_mlflow_tracking_uri_v2 is not supported, called the version of azureml-mlflow in which your planning to introduce get_mlflow_tracking_uri_v2

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_workspace_operations_base.py [hunk 4]
```diff
@@ -86,11 +86,26 @@ def get(self, workspace_name: Optional[str] = None, **kwargs: Any) -> Optional[W
...
+        v2_service_context = {}
+        
+        v2_service_context["subscription_id"] = self._subscription_id
+        v2_service_context["workspace_name"] = workspace_name
+        v2_service_context["resource_group_name"] = resource_group
+        v2_service_context["auth"] = self._credentials
+        
+        from urllib.parse import urlparse
+        parsed_url = urlparse(obj.ml_flow_tracking_uri)
+        host_url = "https://{}".format(parsed_url.netloc)
```
Comment (achauhan-scc) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_workspace_operations_base.py:98]: why https, why not the same scheme of tracking url?

**Review by kingernupur — APPROVED** [review]

**Review by achauhan-scc — APPROVED** [review]

**Pull request merged**